core/src/main/java/org/elasticsearch/bootstrap/JarHell.java

```diff
@@ -198,7 +203,7 @@ public static void checkJarHell(URL urls[]) throws URISyntaxException, IOExcepti
                     public FileVisitResult visitFile(Path file, BasicFileAttributes attrs) throws IOException {
                         String entry = root.relativize(file).toString();
                         if (entry.endsWith(".class")) {
-                            // normalize with the os separator
+                            // normalize with the os separator, remove '.class'
                             entry = entry.replace(sep, ".").substring(0,  entry.length() - 6);
```

Comment:
I wonder if replacing `entry.length() - 6` with `entry.length() - ".class".length()` would be clearer?